MobilityData/mobility-feed-api
# fix: downloaded_at filter

**Summary:**

Changes:
- [x] Fix downloaded_at filtering (remove 2 underscores in the code to make this work) It was not necessary as QA was already working. This was verified with the added unit tests.
- [x] Update downloaded at queries from downloaded_at_gte and downloaded_at_lte to downloaded_before and downloaded_ after
- [x] Includes datetime format in the description of both queries: e.g 2024-02-10T01:06:00Z
- ~In testing steps, make sure that we can do partial matches (e.g just the date with no timestamp)~
- [x] Refactor error handling to standardize status codes and formatting
- [x] Enhance local debugging


Fixes #285 

**Expected behavior:** 


**Testing tips:**

- Start application locally 
- Open Postman hit the following endpoints:
   -  http://localhost:8080/v1/gtfs_feeds/mdb-1/datasets?downloaded_before=2024-12-01T00:00:00Z
   Expect HTTP 200 with body
```
[
{
.. all your 2024 datasets for mdb1 feed
}
]
```
   -  http://localhost:8080/v1/gtfs_feeds/mdb-1/datasets?downloaded_before=2024
   Expect HTTP 422 with body
```
{
    "detail": "Invalid date format for 'downloaded_before'. Expected ISO 8601 format, example: '2021-01-01T00:00:00Z'"
}
```
   -  http://localhost:8080/v1/gtfs_feeds/{id}/datasets?downloaded_after=2024
   Expect HTTP 422 with body
```
{
    "detail": "Invalid date format for 'downloaded_after'. Expected ISO 8601 format, example: '2021-01-01T00:00:00Z'"
}
```
   -  http://localhost:8080/dataset_latitudes=2024-01-30T00:00:00&dataset_longitudes=00&bounding_filter_method=completely_enclosed
   Expect HTTP 422 with body
```
{
    "detail": "Invalid bounding coordinates 2024-01-30T00:00:00 00"
}
```
   -  http://localhost:8080/v1/gtfs_feeds/invalid_id
   Expect HTTP 422 with body
```
{
    "detail": "GTFS feed '2' not found"
}
```
    -  http://localhost:8080/v1/gtfs_rt_feeds/invalid
   Expect HTTP 422 with body
```
{
    "detail": "GTFS realtime Feed 'invalid' not found"
}
```

Please make sure these boxes are checked before submitting your pull request - thanks!
